### PR TITLE
gha: verilator: Add '--locked' to sccache install

### PR DESCRIPTION
**Pull request by wkkuna**
Having run a workflow (first time in a while) in the fork repository I noticed the `Install sccache` step fail to compile `gha-toolkit`.

Link to failed workflow log: https://github.com/antmicro/caliptra-rtl/actions/runs/10059008558/job/27805130656

Adding `--locked` flag to `cargo install` ensures the crates used to build the `sccache` are exactly those that are specified by `Cargo.lock`.

The tool build log with `--locked` flag (note: this workflow may fail as I'm currently testing a new feature in the github flow): https://github.com/antmicro/caliptra-rtl/actions/runs/10072642584/job/27844989866#step:5:1

### CODE REVIEW

**Comment by calebofearth** [comment]
Cool! I had just updated the SCCACHE version to 0.8.1 to get around this in my other PR, which also solves the issue. 
Any reason we want to leave this on 0.3.3?
Either way, the SCCACHE install is also performed in https://github.com/antmicro/caliptra-rtl/blob/wkuna/sccache-install-fix/.github/workflows/interactive-debugging.yml, can you update it there as well?

**Comment by wkkuna** [comment]
Thanks for a quick reply!
Sure thing, added it to `interactive-debugging.yml` as well.

> Any reason we want to leave this on 0.3.3?

None that I know of.

File: .github/workflows/interactive-debugging.yml
Comment (calebofearth) [inline .github/workflows/interactive-debugging.yml:1]: SCCACHE is installed in two places - can you also update line 117?

**Review by calebofearth — APPROVED** [review]

**Comment by calebofearth** [comment]
Ah, looks like there's another workflow issue. You'll need to add
```
     permissions:
      pages: write
      id-token: write
```
in pull_request.yml for the doc_gen job.
For reference, see: https://github.com/chipsalliance/caliptra-rtl/pull/558/files#diff-a0fe23534b616d51ce686d2a1bcd1a78bc75074aef1a2f6ee96c9469991e1a4c

**Review by calebofearth — DISMISSED** [review]

**Comment by calebofearth** [comment]
@wkkuna Thanks for the updates. 
Compile.yml Check is failing - I'm not sure why. Need to debug this.
Hash Check is failing - to fix that you'll need to run [stamp_repo.sh](https://github.com/antmicro/caliptra-rtl/blob/wkuna/sccache-install-fix/.github/scripts/stamp_repo.sh) and commit the changes to `pr_hash` and `pr_timestamp`. This is meant to be an automated step that reflects your branch passing an internal pipeline, but because you're only updating pipeline files it needs to be manual this time. I intend to update this so it's less burdensome in the future. This step needs to be the last commit in the PR, so we need to root-cause and fix the compile.yml check first.

**Comment by calebofearth** [comment]
> @wkkuna Thanks for the updates. Compile.yml Check is failing - I'm not sure why. Need to debug this. Hash Check is failing - to fix that you'll need to run [stamp_repo.sh](https://github.com/antmicro/caliptra-rtl/blob/wkuna/sccache-install-fix/.github/scripts/stamp_repo.sh) and commit the changes to `pr_hash` and `pr_timestamp`. This is meant to be an automated step that reflects your branch passing an internal pipeline, but because you're only updating pipeline files it needs to be manual this time. I intend to update this so it's less burdensome in the future. This step needs to be the last commit in the PR, so we need to root-cause and fix the compile.yml check first.

Just had to rerun the job to get the compile.yml check to pass.
Can you run the stamp-repo script I mentioned? Then this PR should be g2g. Thanks!

**Comment by wkkuna** [comment]
@calebofearth Ran the script locally and committed new hashes.

**Review by calebofearth — DISMISSED** [review]

**Comment by calebofearth** [comment]
> @calebofearth Ran the script locally and committed new hashes.

Looks like the committed hash is incorrect. Do you have local file modifications where you checked out this branch? 
I was able to clone your branch to my area and produce the expected hash, so either you have local edits to your branch or the script is behaving differently in your environment for some reason. Can you retry it with a pristine checkout?

**Comment by wkkuna** [comment]
Sorry, I didn't realize. Checked `file_hash.sh` locally, should be ok now.

**Review by calebofearth — APPROVED** [review]

**Review by Nitsirks — APPROVED** [review]

**Review by nileshbpat — APPROVED** [review]

**Pull request merged**